### PR TITLE
frontend-monorepo-127 Generalize stats page to work for all environments

### LINKED ISSUE #127: Generalize stats page to work for all environments

### PR DESCRIPTION
**Pull request by sam-keen**
Closes #127

### CODE REVIEW

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx [hunk 1]
```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
...
+    title: 'Stagnet 1',
+    stats: 'https://n03.s.vega.xyz/datanode/rest/statistics',
+    nodes: 'https://n03.s.vega.xyz/datanode/rest/nodes-data',
+  },
+  stagnet2: {
+    title: 'Stagnet 2',
+    stats: 'https://n01.stagnet2.vega.xyz/datanode/rest/statistics',
+    nodes: 'https://n01.stagnet2.vega.xyz/datanode/rest/nodes-data',
+  },
+};
```
Comment (dexturr) [inline libs/network-stats/src/components/stats-manager/stats-manager.tsx:49]: chore (blocking): can't hard code this. We can add environments or remove them at any point, we don't want to have to deploy the frontend if an evironment is changed. You should use environment variables here, one fo the REST url (bear in mind for mainnet this could be anything) and a network (which also could be anything).

The API paths are consistent so you need:
```
const title = process.env['NX_VEGA_ENV']
const endpoint = process.env['NX_VEGA_REST']
return {
  title,
  stats: new URL(endpoint, '/statistics').href
  nodes: new URL(endpoint, '/nodes-data').href
}
```

Then adjust the different environment files

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx [hunk 2]
```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
...
+      return dataSources.stagnet1;
+    }
+
+    if (dataNodeUrl.includes('stagnet2')) {
+      return dataSources.stagnet2;
+    }
+  }
+
+  return dataSources.mainnet;
+};
```
Comment (dexturr) [inline libs/network-stats/src/components/stats-manager/stats-manager.tsx:77]: chore (blocking): we cannot detect network like this. Use the environement variable as described above with a map and a default of "unknown"

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx [hunk 3]
```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
...
+
 export const StatsManager = ({ className }: statsManagerProps) => {
+  const dataSource = useRef<dataSourceProps>(getDataSource());
```
Comment (dexturr) [inline libs/network-stats/src/components/stats-manager/stats-manager.tsx:80]: chore (blocking):
```suggestion
  const dataSource = useState<dataSourceProps>(getDataSource());
```

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx [hunk 4]
```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
...
   const [data, setData] = useState<IStructuredStats | null>(null);
   const [error, setError] = useState<Error | null>(null);
 
   useEffect(() => {
+    getDataSource();
```
Comment (dexturr) [inline libs/network-stats/src/components/stats-manager/stats-manager.tsx:85]: queston (blocking): why is this needed?

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx (hunk 1 above)
Comment (dexturr) [reply libs/network-stats/src/components/stats-manager/stats-manager.tsx:49]: Also this component shouldn't care about vega environments ect. hoist this logic to the consumer, take these variables as props to the components I think makes the most sense.

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx [hunk 5]
```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
   className?: string;
 }
 
+interface dataSourceProps {
```
Comment (dexturr) [inline libs/network-stats/src/components/stats-manager/stats-manager.tsx:17]: chore (blocking): interfaces should e PascalCase

File: apps/explorer/src/app/routes/home/index.tsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { StatsManager } from '@vegaprotocol/mainnet-stats-manager';
+import { StatsManager } from '@vegaprotocol/network-stats';
```
Comment (dexturr) [inline apps/explorer/src/app/routes/home/index.tsx:1]: question (blocking): why is the package called network-stats when directory is called stats?

**Review by dexturr — CHANGES_REQUESTED** [review]
The environment detection logic is not suitable

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx (hunk 4 above)
Comment (sam-keen) [reply libs/network-stats/src/components/stats-manager/stats-manager.tsx:85]: Ah that's an error - that was my first attempt and should have been moved when I switched to getting the data in the `useRef`

File: apps/explorer/src/app/routes/home/index.tsx (hunk 6 above)
Comment (sam-keen) [reply apps/explorer/src/app/routes/home/index.tsx:1]: urgh - naming is hard! I considered calling the package the same thing as the app, because the app is essentially just a wrapper around the component. But that didn't feel like a great idea, so opted for something that was different but the same, if that makes sense :)

File: libs/network-stats/src/components/stats-manager/stats-manager.tsx (hunk 1 above)
Comment (sam-keen) [reply libs/network-stats/src/components/stats-manager/stats-manager.tsx:49]: Great points, thanks

**Review by dexturr — APPROVED** [review]

**Review by bglownia — APPROVED** [review]

**Review by edd — APPROVED** [review]
All looks fine to me, but I am new to NX. Luckily @bglownia has approved already!
